src/Microsoft.Identity.Abstractions/ManagedIdentity/ManagedIdentityDescription.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.ComponentModel;
+
+namespace Microsoft.Identity.Abstractions
+{
+    /// <summary>
+    /// Data object to hold the description of a managed identity for an application to use for authentication.
+    /// </summary>
+    public class ManagedIdentityDescription
+    {
+        /// <summary>
+        /// Gets or sets the type of managed identity (and if applicable the unique id) as defined in
+        /// <see cref="ManagedIdentityType"/>.Defaults to <see cref="ManagedIdentityType.SystemAssigned"/> if not set.
+        /// </summary>
+        [DefaultValue(ManagedIdentityType.SystemAssigned)]
+        public ManagedIdentityType IdType { get; set; }
+
+
+        /// <summary>
+        /// Gets or sets the value of the unique id correlated to the <see cref="IdType"/> field for the
+        /// managed identity resource being described. Only used when <see cref="IdType"/> is set to a user-assigned option.
+        /// </summary>
+        public string? IdValue { get; set; }
```

Comment:
>IdValue [](http://example.com/codeflow?start=23&length=7)

What is `IdValue`? Is this a client ID? Object ID?
Seems like we have deviated from the original design from the GitHub issue, was that the intent? Seems less straightforward and more complicated.

Reply:
If you look at the ManagedIdentityType enum the client Id is specified there for the user-assigned identities. I originally had this as two enums one for system/user-assigned and one for UniqueIdType but @jmprieur preferred a less verbose definition so we flattened them.

It deviates from the original design on purpose (I did discuss with JM) as the original design would have left too much room for misconfiguration and would also cause a messier implementation in Id Web. This will be harder to get wrong and easier to check.

Reply:
design changed, no longer relevant